habitat_baselines/motion_planning/mp_sim.py

```diff
@@ -0,0 +1,508 @@
+        Captures the environment state from the depth camera and forms a point
+        cloud.
+        """
+        take_count = 2000
+        sphere_radius = 0.05
+        mass = 0
+
+        if use_prev and len(self.render_bodies) != 0:
+            return
+
+        for body_id in self.render_bodies:
+            p.removeBody(body_id, physicsClientId=self.pc_id)
+        self.render_bodies = []
+
+        if not use_prev:
+            # We are in a position where we CAN retract the arm.
+            orig_arm = self._sim.get_arm_pos()
+            self._sim.hack_retract_arm()
+
+        obs = self._sim.step(0)
+
+        depth = obs["depth"]
+        voxel_mapper = VoxelMapper(False)
+        voxel_mapper.compute_voxels(depth, self._sim)
+        voxels = voxel_mapper.get_voxels()
+        random.shuffle(voxels)
+        voxels = np.array(voxels)
+        voxels = voxels[:take_count, :3]
+
+        if not use_prev:
+            self._sim.robot.arm_joint_pos = orig_arm
+
+        sphere_id = p.createCollisionShape(
+            p.GEOM_SPHERE, radius=sphere_radius, physicsClientId=self.pc_id
+        )
+        # sphere_id = -1
+        sphere_reg = p.createVisualShape(
+            shapeType=p.GEOM_SPHERE,
+            rgbaColor=[1, 0, 0, 1],
+            radius=sphere_radius,
+            physicsClientId=self.pc_id,
+        )
+
+        T = self._sim.get_robot_transform()
+        self.robot_transform = T
+        rot = mn.Quaternion.from_matrix(T.rotation())
+        rot = [*rot.vector, rot.scalar]
+        pos = list(T.translation)
+
+        js = self._sim.get_joints_pos()
+
+        self.fix_joint = js
+
+        p.resetBasePositionAndOrientation(
+            self.robo_id, pos, rot, physicsClientId=self.pc_id
+        )
+
+        p.setGravity(0, 0, 0, physicsClientId=self.pc_id)
+
+        self._set_hab_js(js)
+
+        for voxel in voxels:
+            voxel_id = p.createMultiBody(
+                mass, sphere_id, sphere_reg, voxel, physicsClientId=self.pc_id
+            )
+            self.render_bodies.append(voxel_id)
+
+    def create_viz(self, viz_pos, r=0.05):
+        sphere_id = p.createCollisionShape(
+            p.GEOM_SPHERE, radius=r, physicsClientId=self.pc_id
+        )
+        sphere_targ = p.createVisualShape(
+            shapeType=p.GEOM_SPHERE,
+            rgbaColor=[0, 0, 1, 1],
+            radius=r,
+            physicsClientId=self.pc_id,
+        )
+        return p.createMultiBody(
+            0.0, sphere_id, sphere_targ, viz_pos, physicsClientId=self.pc_id
+        )
+
+    def get_robot_transform(self):
+        return self.robot_transform
+
+    def get_collisions(self, count_obj_colls, ignore_names, verbose):
+        colls = p.getContactPoints(physicsClientId=self.pc_id)
+
+        def convert_coll(x):
+            linkA, bodyA = p.getBodyInfo(x[1], physicsClientId=self.pc_id)
+            linkB, bodyB = p.getBodyInfo(x[2], physicsClientId=self.pc_id)
+            bodyA = bodyA.decode("ascii")
```

Comment:
Hmm, we may want to revisit the API to prevent the need to decode here. It's kinda off for a Python object and easy to fix in PyBind